src/queueFsm.js

```diff
@@ -151,14 +151,14 @@ var Factory = function (options, connection, topology, serializers, queueFn) {
     },
 
     check: function () {
-      var deferred = defer();
+      const deferred = defer();
       this.handle('check', deferred);
       return deferred.promise;
     },
 
     purge: function () {
       return new Promise(function (resolve, reject) {
-        var _handlers;
+        let _handlers = null;
```

Comment:
I set these to null explicitly so we could leave them as let. The `_handlers` variable is used in the functions below this line but still above its instantiation. So I like having it defined here, but that means we can't use const.

Any thoughts on this approach or alternatives?

(There's 2 more like this)

Reply:
This looks like a good solution to the given scenario